crates/sys/src/lib.rs

```diff
@@ -29,12 +29,18 @@ mod inlines;
 )]
 mod bindings;
 
-/// Create a static C string.
+/// Create a static C string of type [`&'static CStr`][`core::ffi::CStr`].
 /// Will automatically add a NUL terminator.
+pub use cstr::cstr;
+
+/// Create a static C string of type [`*const c_char`][core::ffi::c_char]
+/// referring to a `'static` string.
+/// Will automatically add a NUL terminator.
+// TODO: don't produce intermediate `CStr` whose `length` part we don't use
 #[macro_export]
 macro_rules! c_string {
     ($str:expr $(,)?) => {{
-        concat!($str, "\0").as_ptr() as *const core::ffi::c_char
+        $crate::cstr!($str).as_ptr()
```

Comment:
Something about this change broke the logging macros. From CI (compiling the examples):
```
error: unexpected token
  --> flipperzero/examples/hello-rust.rs:31:5
   |
31 |     info!("Hello, reader of the logs!");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::log` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected token
  --> flipperzero/examples/hello-rust.rs:35:5
   |
35 |     debug!("Return code: {}", ret_code);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::log` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `flipperzero` (example "hello-rust") due to 2 previous errors
```

Reply:
I took me some time to find the issue:

It seems that the (imported) `cstr` macro requires the value to be literal (this is logical) which is not true when using `module_path!`, unlike our old `c_string` which (I guess, magically) correctly expands `concat!(module_path!(..), "\0")`.

My suggestion here is to keep the changes, since they are less error-prone due to strictire compile-time validations but provide our convenient macros for things like `module_path` producing valid ptrs, although I am mnot as sure if it is the right approach.

Reply:
I've pushed my vision of this change which reworks `log` macro, although there may be other ways to do it

Reply:
I don't see any relevant changes pushed to this PR.

Reply:
My bad, just pushed the change in 13cca7ae9ead9e1212c270d3c0ae2c8af7b34d42